manifests/agent/windows.pp

```diff
@@ -1,9 +1,14 @@
-class distelli::agent::windows (
-  $version = '3.66.33',
+# Install the Puppet Pipelines agent on Windows
+#
+# Do not use directly; use pipelines::agent.
+class pipelines::agent::windows (
   $tempdir = 'C:/Users/distelli/AppData/Local/Temp/1',
-){
+) {
+  $version = $pipelines::agent::version
+  $workdir = 'C:/Program Files/Distelli'
```

Comment:
Dangerous Default.  Do not hardcode this

Reply:
SystemDrive doesn't have to `C:\`.  `Program Files` can change (32 v 64bit, localisation)

Instead use the `%PROGRAM FILES%` env var as a fact.